src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,287 @@ describe(__filename, () => {
+      [
+        { path: file1, status: 'M' },
+        { path: 'fileWithoutChange', status: '' },
+        { path: file2, status: 'M' },
+      ],
+      { id: headVersionId },
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      { selected_file: file1, diff: fakeExternalDiff },
+    ) as ExternalVersionWithDiff;
+
+    const {
+      _fetchDiff,
+      _fetchVersionFile,
+      dispatchSpy,
+      fakeThunk,
+    } = loadDiffAndRender({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      version,
+      buildTree: false,
+    });
+
+    expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchDiff).not.toHaveBeenCalledWith({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      path: file2,
+    });
+    expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+      addonId,
+      versionId: headVersionId,
+      path: file2,
+    });
+  });
+
+  it.each([['versionFile', 'loadVersionFile'], ['diff', 'loadDiff']])(
+    'does not dispatch fetchDiff or fetchVersionFile for the next file when the current %s is not loaded',
+    (data, action) => {
+      const addonId = 10;
+      const baseVersionId = 44;
+      const headVersionId = 55;
+      const file1 = 'file1.json';
+      const file2 = 'file2.json';
+      const version = createExternalVersionWithEntries(
+        [
+          { path: file1, status: 'M' },
+          { path: 'fileWithoutChange', status: '' },
+          { path: file2, status: 'M' },
+        ],
+        { id: headVersionId },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        { selected_file: file1, diff: fakeExternalDiff },
+      ) as ExternalVersionWithDiff;
+
+      const { _fetchDiff, _fetchVersionFile } = loadDiffAndRender({
+        [action]: false,
+        addonId,
+        baseVersionId,
+        buildTree: true,
+        headVersionId,
+        version,
+      });
+
+      expect(_fetchDiff).not.toHaveBeenCalledWith({
+        addonId,
+        baseVersionId,
+        headVersionId,
+        path: file2,
+      });
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        versionId: headVersionId,
+        path: file2,
+      });
+    },
+  );
+
+  it('does not dispatch fetchVersionFile for the next file if it is loading', () => {
+    const addonId = 10;
+    const baseVersionId = 44;
+    const headVersionId = 55;
+    const file1 = 'file1.json';
+    const file2 = 'file2.json';
+    const fakeThunk = createFakeThunk();
+    const _fetchVersionFile = fakeThunk.createThunk;
+    const store = configureStore();
+    const version = createExternalVersionWithEntries(
+      [
+        { path: file1, status: 'M' },
+        { path: 'fileWithoutChange', status: '' },
+        { path: file2, status: 'M' },
+      ],
+      { id: headVersionId },
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      { selected_file: file1, diff: fakeExternalDiff },
```

Comment:
These tests could benefit from a helper that abstracts some things like setting up `store`, `diff`, `_fetchVersionFile`, dispatching `buildTree`, etc. Similar to in the Browse patch, just beware of letting the tests control the inputs that are necessary for the assertions. I see that some tests need to control a lot of inputs which is fine.

Reply:
To reuse the existing helper `loadDiffAndRender` in this test, I passed a function [`moreSetup`](https://github.com/mozilla/addons-code-manager/pull/1085/files#diff-549d8d8730ca2dc5ecce96cdb68f5020R183) as parameter to [dispatch one more actions before rendering](https://github.com/mozilla/addons-code-manager/pull/1085/files#diff-549d8d8730ca2dc5ecce96cdb68f5020R1054-R1070). But I am not very sure this is a proper way.